src/crypto/crypto.h

```diff
@@ -68,71 +67,78 @@ namespace crypto
 {
 
 #pragma pack(push, 1)
-POD_CLASS ec_point
+struct ec_point
 {
 	char data[32];
 };
 
-POD_CLASS ec_scalar
+struct ec_scalar
 {
 	char data[32];
 };
 
-POD_CLASS public_key : ec_point
+struct public_key : ec_point
 {
 	friend class crypto_ops;
 };
 
-POD_CLASS scalar_16
+struct scalar_16
 {
 	uint8_t data[16];
 };
 
 using secret_key_16 = tools::scrubbed<scalar_16>;
 using secret_key = tools::scrubbed<ec_scalar>;
 
-POD_CLASS public_keyV
+struct public_keyV
 {
 	std::vector<public_key> keys;
 	int rows;
 };
 
-POD_CLASS secret_keyV
+struct secret_keyV
 {
 	std::vector<secret_key> keys;
 	int rows;
 };
 
-POD_CLASS public_keyM
+struct public_keyM
 {
 	int cols;
 	int rows;
 	std::vector<secret_keyV> column_vectors;
 };
 
-POD_CLASS key_derivation : ec_point
+struct key_derivation : ec_point
 {
 	friend class crypto_ops;
 };
 
-POD_CLASS key_image : ec_point
+struct key_image : ec_point
 {
 	friend class crypto_ops;
 };
 
-POD_CLASS signature
+struct signature
 {
 	ec_scalar c, r;
 	friend class crypto_ops;
 };
+
+// New payment id system
+struct uniform_payment_id
+{
+	uint64_t zero = 0; // Zero field needs to equal zero for the structure to be valid
```

Comment:
why is the default not `-1` and only set to zero if a valid pid is added?

Reply:
That's probably a better way of doing it - will change it.